src/main/java/org/ethereum/beacon/discovery/schema/NodeSession.java

```diff
@@ -74,7 +74,8 @@ public NodeSession(
       NodeBucketStorage nodeBucketStorage,
       AuthTagRepository authTagRepo,
       Consumer<NetworkParcel> outgoingPipeline,
-      Random rnd) {
+      Random rnd,
+      ExpirationSchedulerFactory expirationSchedulerFactory) {
```

Comment:
nit: Would it make more sense to do a dependency injection instead here? 
```suggestion
      ExpirationScheduler expirationScheduler) {
```

Reply:
Yeah it probably does in that case - good spot.  Generally I've tried to be fairly low touch since the design was built with expiration schedulers being specific to a particular case, so just aimed to ensure that a single `ScheduledExecutorService` was shared between them all.